src/solver/optim/ortoolsImpl/linearProblem.cpp

```diff
@@ -0,0 +1,155 @@
+    }
+
+    auto* mpVar = mpSolver_->MakeVar(lb, ub, integer, name);
+    auto mipVar = std::make_unique<OrtoolsMipVariable>(mpVar);
+
+    if (!mpVar || !mipVar)
+    {
+        logs.error() << "Couldn't add variable to Ortools MPSolver: " << name;
+    }
+
+    if (!variables_.try_emplace(name, std::move(mipVar)).second)
+    {
+        logs.error() << "Error adding variable: " << name;
+    }
+
+    return variables_[name].get();
+}
+
+Api::MipVariable* OrtoolsLinearProblem::addNumVariable(double lb,
+                                                       double ub,
+                                                       const std::string& name)
+{
+    return addVariable(lb, ub, false, name);
+}
+
+Api::MipVariable* OrtoolsLinearProblem::addIntVariable(double lb,
+                                                       double ub,
+                                                       const std::string& name)
+{
+    return addVariable(lb, ub, true, name);
+}
+
+Api::MipVariable* OrtoolsLinearProblem::getVariable(const std::string& name)
+{
+    return variables_[name].get();
+}
+
+Api::MipConstraint* OrtoolsLinearProblem::addConstraint(double lb,
+                                                        double ub,
+                                                        const std::string& name)
+{
+    if (constraints_.contains(name))
+    {
+        logs.error() << "This constraint already exists: " << name;
+        throw std::bad_function_call();
+    }
+
+    auto* mpConstraint = mpSolver_->MakeRowConstraint(lb, ub, name);
+    auto mipConstraint = std::make_unique<OrtoolsMipConstraint>(mpConstraint);
+
+    if (!constraints_.try_emplace(name, std::move(mipConstraint)).second)
+    {
+        logs.error() << "Error adding constraint: " << name;
+    }
+
+    return constraints_[name].get();
+}
+
+Api::MipConstraint* OrtoolsLinearProblem::getConstraint(const std::string& name)
+{
+    return constraints_[name].get();
+}
+
+void OrtoolsLinearProblem::setObjectiveCoefficient(Api::MipVariable* var, double coefficient)
+{
+    objective_->setCoefficient(var, coefficient);
+}
+
+double OrtoolsLinearProblem::getObjectiveCoefficient(Api::MipVariable* var)
+{
+    return objective_->getCoefficient(var);
+}
+
+void OrtoolsLinearProblem::setMinimization()
+{
+    objective_->setMinimization();
+}
+
+void OrtoolsLinearProblem::setMaximization()
+{
+    objective_->setMaximization();
+}
+
+Api::MipSolution* OrtoolsLinearProblem::solve()
+{
+    mpSolver_->EnableOutput();
+    auto status = mpSolver_->Solve(*param_);
+
+    std::map<std::string, std::pair<Api::MipVariable*, double>> solution;
+    for (auto& var: mpSolver_->variables())
+    {
+        auto pair = std::make_pair(variables_.at(var->name()).get(), var->solution_value());
+        solution.try_emplace(var->name(), pair);
+    }
+
+    auto* mpObjective = dynamic_cast<OrtoolsMipObjective*>(objective_.get());
+    double objectiveValue = mpObjective->getMpObjective()->Value();
```

Comment:
Interface leak. If you only need the objective value, then add a `double value()` virtual method to the interface.